File: 4-MedianOfTwoSortedArrays/4-MedianOfTwoSortedArrays.java
// Last updated: 11/05/2026, 01:34:45
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        double median;
          int m = nums1.length;
          int n = nums2.length;
          int arr[] = new int[n+m];
          for (int i = 0; i < m; ++i)
          {
            arr[i] = nums1[i];
          }
          for (int i = 0; i < n; ++i)
          {
            arr[m + i] = nums2[i];
          }
          Arrays.sort(arr);
          if ((m + n) % 2 == 0)
          {
            int f = (m + n) / 2;
            int s = (m + n) / 2 + 1;
            median = (arr[f] + arr[s - 2]) / 2.0;
          }
          else
          {
            int mid = (m + n) / 2;
            median = arr[mid];
          }
          return median;
    }
}

File: 4-MedianOfTwoSortedArrays/4-MedianOfTwoSortedArrays.cpp
// Last updated: 11/05/2026, 01:34:41
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        double result;
        vector<double> mergedArray;
        int i = 0; int j = 0;
        while (i < nums1.size() && j < nums2.size()){
            if (nums1[i] <= nums2[j]){ 
                mergedArray.push_back(nums1[i]);
                i++;
            }
            else{
                mergedArray.push_back(nums2[j]);
                j++;
            }
        }
        while (i < nums1.size()){
            mergedArray.push_back(nums1[i]);
            i++;
        }
        while (j < nums2.size()){
            mergedArray.push_back(nums2[j]);
            j++;
        }
        if ((nums1.size() + nums2.size()) %2 == 0){
            int index = floor((nums1.size() + nums2.size())/2);
            result = (mergedArray[index] + mergedArray[index - 1])/2;
        }
        else{
            int index = floor((nums1.size() + nums2.size())/2);
            result = mergedArray[index];
        }
        return result;
    }
};